api/messages.py

```diff
@@ -0,0 +1,269 @@
+        super().__init__(*args, **kwargs)
+        self.message = self.response.body
+
+class AddedMessage(GenericResponse):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.message = self.response.body
+
+class GetMessagesCount(GenericTransaction):
+    '''
+    Get the count of messages
+    '''
+    def __init__(self):
+        self.request = Request(Endpoint.MESSAGES, Method.GET,
+            {
+                "category": "message",
+                "count": True
+            })
+
+    def setResponse(self, response: Response):
+        self.response = MessagesCount(response)
+
+
+class GetMessagesWithOffsetAndLimit(GenericTransaction):
+    '''
+    Retrieve messages with offset and limit
+    '''
+    def __init__(self, offset: int, limit: int):
+        self.request = Request(Endpoint.MESSAGES, Method.GET,
+            {
+                "category": "message",
+                "offset": offset,
+                "limit": limit
+            })
+
+    def setResponse(self, response: Response):
+        self.response = Messages(response)
+
+
+class GetMessageById(GenericTransaction):
+    '''
+    Retrieve message by message ID
+    '''
+    def __init__(self, messageID: int):
+        self.request = Request(Endpoint.MESSAGES, Method.GET,
+            {
+                "category": "message",
+                "messageID": messageID
+            })
+
+    def setResponse(self, response: Response):
+        self.response = MessageByIdEntry(response)
+
+class GetMessageByThreadIdWithLimit(GenericTransaction):
+    '''
+    Retrieve messages by thread ID with limit
+    '''
+    def __init__(self, threadID: int, limit: int):
+        self.request = Request(Endpoint.MESSAGES, Method.GET,
+            {
+                "category": "message",
+                "threadID": threadID,
+                "offset":0,
+                "limit":limit
+            })
+
+    def setResponse(self, response: Response):
+        self.response = Messages(response)
+
+class AddMessage(GenericTransaction):
+    '''
+    Send a message to specified phone number
+    '''
+    def __init__(self, number: str, messageBody: str):
+        self.request = Request(Endpoint.MESSAGES, Method.POST,
+            {
+                "category": "message",
+                "number": number,
+                "messageBody": messageBody
+            })
+
+    def setResponse(self, response: Response):
+        self.response = AddedMessage(response)
+
+class DeleteMessageById(GenericTransaction):
+    '''
+    Delete message specified by Id
+    '''
+    def __init__(self, messageID: int):
+        self.request = Request(Endpoint.MESSAGES, Method.DEL,
+            {
+                "category": "message",
+                "messageID": messageID
+            })
+
+    def setResponse(self, response: Response):
+        self.response = response
+
+'''
+Templates
+'''
+
+class TemplatesCount(GenericResponse):
```

Comment:
please add minimal docs to the classes, not over them :) if you add docs at the top of the module it will be module docstring, it added under the class name - then it will be the class one. In the middle it doesn't add value

Reply:
Removed redundant comments